### PR TITLE
modified auth utils

### PR DESCRIPTION
**Pull request by 1298se**

### CODE REVIEW

File: contrail-ui/src/utils/auth-utils.ts [hunk 1]
```diff
@@ -18,61 +21,75 @@ export function initializeFirebase() {
...
 
 /**
  * Registers a user to Firebase. *NOTE*: {@link initializeFirebase} must be called before.
  * @param displayName the user's display name received from registration
  * @param email the user's email received from registration.
  * @param password the user's password received from registration.
- *
+ * @returns a promise that resolves with the current user, or rejects with the error
  */
 export function registerUser(displayName: string, email: string, password: string) {
```
Comment (notoriousmango) [inline contrail-ui/src/utils/auth-utils.ts]: export function registerUser(displayName: string, email: string, password: string) : Promise<${type of user}>

File: contrail-ui/src/utils/auth-utils.ts [hunk 2]
```diff
@@ -18,61 +21,75 @@ export function initializeFirebase() {
...
     });
 }
 
 /**
  * Logins a user to Firebase. *NOTE* {@link initializeFirebase} must be called before.
  * @param email the user's email received from login
  * @param password the user's password received from login
+ * @return a promise that resolves with the current user, or rejects with the error
  */
 export function loginUser(email: string, password: string) {
```
Comment (notoriousmango) [inline contrail-ui/src/utils/auth-utils.ts]: export function loginUser(email: string, password: string) : Promise<${type of user}> {

File: contrail-ui/src/utils/auth-utils.ts [hunk 3]
```diff
@@ -18,61 +21,75 @@ export function initializeFirebase() {
...
     console.log("logging in user");
-    firebase.auth().signInWithEmailAndPassword(email, password).then(function() {
-        console.log("login successful");
-    }).catch(function(error) {
-        console.log("login failed: ", error);
+    return new Promise((resolve, reject) => {
+        firebase.auth().signInWithEmailAndPassword(email, password).then(() => {
+            console.log("login successful");
+            const user = firebase.auth().currentUser;
+            resolve(user);
```
Comment (notoriousmango) [inline contrail-ui/src/utils/auth-utils.ts]: resolve(firebase.auth().currentUser);

File: contrail-ui/src/utils/auth-utils.ts [hunk 4]
```diff
@@ -18,61 +21,75 @@ export function initializeFirebase() {
...
-        return null;
-    }
-    user.getIdToken(true).then(function(idToken) {
-        return idToken;
-      }).catch(function(error) {
-        return null;
-      });
+    return new Promise((resolve, reject) => {
+        const user = firebase.auth().currentUser;
+        if (user == null) {
```
Comment (notoriousmango) [inline contrail-ui/src/utils/auth-utils.ts]: user === null

File: contrail-ui/src/utils/auth-utils.ts [hunk 5]
```diff
@@ -18,61 +21,75 @@ export function initializeFirebase() {
...
+            resolve(user);
```
Comment (notoriousmango) [inline contrail-ui/src/utils/auth-utils.ts]: resolve null

File: contrail-ui/src/utils/auth-utils.ts [hunk 6]
```diff
@@ -18,61 +21,75 @@ export function initializeFirebase() {
...
     });
 }
 
 /**
  * Gets the ID Token of a current user, or null
- * @returns current user ID Token, or null
+ * @returns a promise that resolves null if there is no current user or the user's IDToken if there is,
+ *  or rejects with error.
  */
 export function getUserToken() {
```
Comment (notoriousmango) [inline contrail-ui/src/utils/auth-utils.ts]: export function getUserToken() : string {

File: contrail-ui/src/utils/auth-utils.ts (hunk 3 above)
Comment (notoriousmango) [reply contrail-ui/src/utils/auth-utils.ts]: doesn't rly matter its up to you

Comment (1298se) [reply contrail-ui/src/utils/auth-utils.ts]: added null check

File: contrail-ui/src/utils/auth-utils.ts (hunk 6 above)
Comment (notoriousmango) [reply contrail-ui/src/utils/auth-utils.ts]: means the function returns the promise that can be either stirng or null

File: contrail-ui/src/utils/auth-utils.ts (hunk 2 above)
Comment (1298se) [reply contrail-ui/src/utils/auth-utils.ts]: done

File: contrail-ui/src/utils/auth-utils.ts (hunk 1 above)
Comment (1298se) [reply contrail-ui/src/utils/auth-utils.ts]: done

File: contrail-ui/src/utils/auth-utils.ts (hunk 6 above)
Comment (1298se) [reply contrail-ui/src/utils/auth-utils.ts]: Done

File: contrail-ui/src/utils/auth-utils.ts (hunk 4 above)
Comment (1298se) [reply contrail-ui/src/utils/auth-utils.ts]: done

File: contrail-ui/src/utils/auth-utils.ts [hunk 7]
```diff
@@ -18,61 +21,79 @@ export function initializeFirebase() {
...
-        }
-        user.updateProfile({
-            displayName,
-        }).then(function() {
-            console.log("profile update successful");
+    return new Promise((resolve, reject) => {
+        firebase.auth().createUserWithEmailAndPassword(email, password).then(() => {
+            const user = firebase.auth().currentUser;
+            if (user === null) {
+                console.log("registration failed: user is null");
```
Comment (notoriousmango) [inline contrail-ui/src/utils/auth-utils.ts]: console.error

File: contrail-ui/src/utils/auth-utils.ts [hunk 8]
```diff
@@ -18,61 +21,79 @@ export function initializeFirebase() {
...
+                resolve(null);
+            } else {
+                console.log("registration successful: updating profile");
+                user.updateProfile({
+                    displayName,
+                }).then(() => {
+                    console.log("profile updat successful");
+                    resolve(user);
+                }).catch((error) => {
+                    console.log("profile update failed:  ", error);
```
Comment (notoriousmango) [inline contrail-ui/src/utils/auth-utils.ts]: console.error

File: contrail-ui/src/utils/auth-utils.ts [hunk 9]
```diff
@@ -18,61 +21,79 @@ export function initializeFirebase() {
...
+        firebase.auth().signInWithEmailAndPassword(email, password).then(() => {
+            console.log("login successful");
+            const user = firebase.auth().currentUser;
+            if (user === null) {
+                resolve(null);
+            } else {
+                resolve(user);
+            }
+        }).catch((error) => {
+            console.log("login failed: ", error);
```
Comment (notoriousmango) [inline contrail-ui/src/utils/auth-utils.ts]: console.error

File: contrail-ui/src/utils/auth-utils.ts (hunk 7 above)
Comment (1298se) [reply contrail-ui/src/utils/auth-utils.ts]: done

File: contrail-ui/src/utils/auth-utils.ts (hunk 9 above)
Comment (1298se) [reply contrail-ui/src/utils/auth-utils.ts]: done

File: contrail-ui/src/utils/auth-utils.ts (hunk 8 above)
Comment (1298se) [reply contrail-ui/src/utils/auth-utils.ts]: done

**Pull request merged**